packages/ui-components/src/lib/providers/registry/RegistryManager.ts

```
@@ -0,0 +1,148 @@
+		try {
+			const url = new URL(window.location.href);
+			if (value) {
+				url.searchParams.set(RegistryManager.STORAGE_KEY, value);
+			} else {
+				url.searchParams.delete(RegistryManager.STORAGE_KEY);
+			}
+			window.history.pushState({}, '', url.toString());
+		} catch (error) {
+			throw new Error(
+				'Failed to update URL parameter: ' +
+					(error instanceof Error ? error.message : String(error))
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)